src/components/Attachments/AttachmentCarousel/index.native.js

```diff
@@ -138,16 +147,17 @@ function AttachmentCarousel({report, reportActions, parentReportActions, source,
 
                             <AttachmentCarouselPager
                                 items={attachments}
+                                scrollEnabled={isZoomedOut}
                                 renderItem={renderItem}
                                 initialIndex={page}
-                                onPageSelected={({nativeEvent: {position: newPage}}) => updatePage(newPage)}
-                                onPinchGestureChange={(newIsPinchGestureRunning) => {
-                                    setIsPinchGestureRunning(newIsPinchGestureRunning);
-                                    if (!newIsPinchGestureRunning && !shouldShowArrows) {
-                                        setShouldShowArrows(true);
+                                onTap={() => {
+                                    if (!isZoomedOut) {
+                                        return;
                                     }
+                                    setShouldShowArrows(!shouldShowArrows);
                                 }}
-                                onSwipeDown={onClose}
+                                onPageSelected={({nativeEvent: {position: newPage}}) => updatePage(newPage)}
+                                onScaleChanged={handleScaleChange}
```

Comment:
Let's keep this as is, because there is another PR overhauls the context and improves the code a lot: https://github.com/Expensify/App/pull/34080

Reply:
also, we'll want to avoid prop-drilling through all the `AttachmenView`, `AttachmenViewImage` and `ImageView` components. They don't necessarily need this prop